docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,217 @@
+git remote -v
+```
+**origin** 은 나의 저장소, **upstream** 은 *lunchclass/absolute* 여야 합니다.
+```
+origin	https://github.com/hyungheo/absolute.git (fetch)
+origin	https://github.com/hyungheo/absolute.git (push)
+upstream	https://github.com/lunchclass/absolute.git (fetch)
+upstream	https://github.com/lunchclass/absolute.git (push)
+```
+
+
+5. lunchclass/absolute 의 최신 패치를 업데이트하고 하고 싶으면 upstream 으로 부터 fetch 와 rebase 를 진행합니다.
+```
+git fetch upstream
+git rebase upstream/master
+# 컨틀릭트 파일이 존재하여 리졸브가 필요했을 경우
+git rebase --continue
+```
+
+
+## 이슈등록
+기존 코드에서 버그를 발견하셨거나 새로운 아이디어가 떠오르셨다면 먼저 이슈에 올려주시길 바랍니다. (이슈등록과 관련하여 자세한 내용은 [이슈등록가이드](https://github.com/lunchclass/absolute/blob/master/docs/ISSUE_REPORTING.md) 란을 참조)
+등록한 뒤에는 패치를 준비하시기 이전에 먼저 코드오너와 함께 올리신 이슈에 대해서 이야기를 해보시길 권합니다. 코드오너가 누구인지는 [CODEOWNERS](https://github.com/lunchclass/absolute/blob/master/docs/CODEOWNERS) 를 통해 확인하실 수 있습니다.
+
+## 패치준비
+### 패치사이즈
+패치는 가능하다면 적당히 작은 사이즈로 올려주시는 편이 좋습니다. 지나치게 많은 수정을 한 패치에 포함시키면 리뷰어가 좋은 코드리뷰를 하기가 어려워지며 리뷰과정에서 발견할 수 있는 이슈를 놓칠 확률이 높아집니다. 만약 새로운 구현으로 인해 수정사항이 많은 경우는 패치를 단계별로 작게 나누어서 올리시는 것도 좋은 방법입니다.
+
+예)   
+1. 첫번째 패치에서는 함수들의 껍데기만 먼저 올림
+```TypeScript
+function foo(num: number): number {
+  // Not Implemented
+}
+
+function foo(num: number): number {
+  // Not Implemented
+}
+```
+2. 두번째 패치에서 한 함수의 구현부를 추가
+```TypeScript
+function foo(num: number): number {
+  let ret : number = 0;
+  if( num <= 1 ) {
+    return num;
+  } else if( num > 1 ) {
+    ret =  foo(num-1) + foo(num-2);
+  }
+  return ret;
+}
+
+function foo(num: number): number {
+  // Not Implemented
+}
+```
+
+3. 세번째 패치에서 마지막 함수의 구현부를 추가
+```TypeScript
+function foo(num: number): number {
+  let ret : number = 0;
+  if( num <= 1 ) {
+    return num;
+  } else if( num > 1 ) {
+    ret =  foo(num-1) + foo(num-2);
+  }
+  return ret;
+}
+
+
+function bar(num: number): number {
+  return foo(num);
+}
+```
+
+### 파일추가
+패치가 기존에 없던 새로운 파일추가를 포함하고 있다면 boilerplate notice를 추가하셔야 합니다. 그리고 boilerplate notice에는 반드시 연도가 표시되어야만 합니다. 자세한 내용은 [LICENCE_GUIDE](https://github.com/lunchclass/absolute/blob/master/LICENSE.md) 를 참고하시기 바랍니다.
+
+예)  
+
+```TypeScript
+/*
+ * Copyright (c) 2017 The Absolute Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ function foo(num: number): number {
+```
+
+## 커밋준비
+### 코딩스타일
+패치가 어느정도 준비되었다면 커밋하시기 전에 먼저 코딩스타일이 본 프로젝트의 규칙을 따르고 있는지 확인이 필요합니다.
+absolute는 [Microsoft/TypeScript](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines)의 coding style을 준수합니다. 
```

Comment:
We have our coding style guide. Our guide is extending MS style but not exact MS style.